src/component/handlers/edit/editOnCompositionStart.js

```diff
@@ -18,9 +18,10 @@ var EditorState = require('EditorState');
  * The user has begun using an IME input system. Switching to `composite` mode
  * allows handling composition input and disables other edit behavior.
  */
-function editOnCompositionStart(): void {
+function editOnCompositionStart(e): void {
   this.setRenderGuard();
   this.setMode('composite');
+  this._onCompositionStart(e);
```

Comment:
Modifier.removeRange is called with the e.data value of compositionstart later in the current patch. I wonder what is best, the text is not removed while in composition, it just get potentially updated at the end.

Replace DOM content within the model was what I had in mind. Will check editOnInput why I have time.

I'll be off next week.